test/e2e/es/license_test.go

```diff
@@ -127,3 +129,61 @@ func TestEnterpriseTrialLicense(t *testing.T) {
 
 	test.Sequence(initStepsFn, stepsFn, esBuilder).RunSequential(t)
 }
+
+// TestEnterpriseTrialExtension tests that trial extensions can be successfully applied and take effect.
+func TestEnterpriseTrialExtension(t *testing.T) {
+	if test.Ctx().PrivateKeyPath == "" {
+		// skip this test if the dev private key is not configured e.g. because we are testing a production build
+		t.SkipNow()
+	}
+	privateKeyBytes, err := ioutil.ReadFile(test.Ctx().PrivateKeyPath)
+	require.NoError(t, err)
```

Comment:
aren't we reading the content of that file in `e2e/cmd/run/run.go` already? Can we pass it in the context directly?

Reply:
Remember this runs in a k8s job. `run.go` is settting up that job, copying the key into a secret that is mounted in the job's pod. 

Also context is dumped in logs, it's a test key but still we should not put sensitive data in the context.

Reply:
Oh right, I missed that 👍